app/models/game.rb

```diff
@@ -36,4 +36,27 @@ def other_player(player)
   def pass_turn(player)
     update_attributes(active_player_id: other_player(player))
   end
+
+  def stalemate?
+    king_not_in_check? && available_move?
+  end
+
+  def king_not_in_check?
+    king = pieces.find_by(type: 'King', color: current_color)
+    return true unless king.in_check?
+    false
+  end
+
+  def available_move?
+    all_piece = pieces.where(in_game: true)
```

Comment:
when the `Game.new` are called, it automatically call `populate_board!` (line 8 of game.rb). And Piece has `in_game:` column with default value of `true` implemented. .  And then it loop thru each pieces that are till in game if each pieces have any legal move at all.  I renamed `all_piece` to `active_player_pieces` for clarity.
Please let me know if it makes sense or not.   :)